File: DataPowerSyncDesktop/Models/AppSchema.swift
//
//  AppSchema.swift
//  DataPowerSync
//
//  Created by Michael Martell on 10/5/25.
//

import Foundation
import PowerSync

/// This defines the PowerSync SQLite tables
let powerSyncSchema = Schema(
    tables: [
        Table(
            name: "\(AppConfig.DBprefix)counters",
            columns: [
                .integer("count"),
                .text("owner_id"),
                .text("created_at")
            ]
        )
    ]
)

File: data-power-sync-web/src/library/powersync/AppSchema.ts
import { column, Schema, Table } from '@powersync/web';

// Table matching your actual Supabase table name
const powerSyncCounters = new Table(
  {
    // id column (text) is automatically included
    count: column.integer,
    owner_id: column.text,
    created_at: column.text
  },
  { indexes: {} }
);

export const AppSchema = new Schema({
  power_sync_counters: powerSyncCounters
});

export type Database = (typeof AppSchema)['types'];
export type Counter = Database['power_sync_counters'];